lib/inc/pxp-agent/results_mutex.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef SRC_AGENT_RESULTS_MUTEX_HPP_
+#define SRC_AGENT_RESULTS_MUTEX_HPP_
+
+#include <cpp-pcp-client/util/thread.hpp>
+
+#include <map>
+#include <string>
+#include <memory>
+#include <stdexcept>
+
+namespace PXPAgent {
+
+// NOTE(ale): this class does not provide general synchronizaion
+// functions; it provides a cache for named mutexes where it is
+// assumed that a single actor (the RequestProcessor instance, in
+// practice) is allowed to add and remove mutexes, whereas all actors
+// (including instances of the Transaction Status request handler) can
+// lock the cached mutexes
+
+class ResultsMutex {
+  public:
+    struct Error : public std::runtime_error {
+        explicit Error(std::string const& msg) : std::runtime_error(msg) {}
+    };
+
+    static ResultsMutex& Instance() {
+        static ResultsMutex instance {};
+        return instance;
+    }
+
+    using Mutex = PCPClient::Util::mutex;
+    using Mutex_Ptr = std::shared_ptr<PCPClient::Util::mutex>;
+    using Lock = PCPClient::Util::unique_lock<PCPClient::Util::mutex>;
+    using LockGuard = PCPClient::Util::lock_guard<PCPClient::Util::mutex>;
+
+    // TODO(ale): update this as soon as we have tagged cpp-pcp-client
```

Comment:
Adding the boost::defer_lock const and tagging it; Travis and AppVeyor use cpp-pcp-client 1.0.1, so we currently use the function defined in ResultsMutex, but it's not nice for breaking the PCPClient abstraction.